cli/src/command_error.rs

```diff
@@ -296,12 +296,27 @@ impl From<OpsetEvaluationError> for CommandError {
 impl From<SnapshotError> for CommandError {
     fn from(err: SnapshotError) -> Self {
         match err {
-            SnapshotError::NewFileTooLarge { .. } => {
-                user_error_with_message("Failed to snapshot the working copy", err).hinted(
-                    r#"Increase the value of the `snapshot.max-new-file-size` config option if you
-want this file to be snapshotted. Otherwise add it to your `.gitignore` file."#,
-                )
-            }
+            SnapshotError::NewFileTooLarge {
+                path,
+                size,
+                max_size,
+            } => user_error(format!(
+                "Failed to snapshot the working copy\nThe file '{}' is too large to be \
+                 snapshotted ({} bytes).\nThe maximum size allowed is {} bytes.",
+                path.display(),
+                size.0,
+                max_size.0
+            ))
+            .hinted(format!(
+                "This is to prevent large files from being added on accident. You can fix this \
+                 error by:
+  - Adding the file to `.gitignore`
+  - Run `jj config set --repo snapshot.max-new-file-size \"{}B\"`
+    This will increase the maximum file size allowed for new files, in this repository only.
+  - Run `jj --config-toml 'snapshot.max-new-file-size=\"{}B\"' st`
```

Comment:
nit: maybe better to use integer form `size={}`? The "B" suffix wouldn't improve the readability.